dora_node_hub/dora_gemini_diff_drive_navigation/dora_gemini_diff_drive_navigation/main.py

```diff
@@ -0,0 +1,144 @@
+"""Main entry point for the dora_gemini_diff_drive_navigation package."""
+
+import asyncio
+import os
+import threading
+
+import cv2 as cv
+import numpy as np
+import pyarrow as pa
+from dora import Node
+
+from dora_gemini_diff_drive_navigation.controller import DiffDriveGeminiControl
+
+
+def start_event_loop(loop: asyncio.AbstractEventLoop) -> None:
+    """Starts the asyncio event loop in the background thread."""
+    asyncio.set_event_loop(loop)
+    loop.run_forever()
+
+
+def zero_twist() -> np.ndarray:
+    """Create a zero twist array."""
+    return np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0], dtype=np.float64)
+
+
+def main() -> None:
+    """TODO: Add docstring."""
+    node = Node()
+
+    # Required node configuration
+    command = os.getenv("COMMAND", "")
+    if not command:
+        print("COMMAND environment variable is not set. Using command via input event.")
+    # Optional node configuration
+    model = os.getenv("MODEL", "gemini-2.5-flash")
+
+    # Initialize the controller with the specified model.
+    controller = DiffDriveGeminiControl(model=model)
+    last_image = None
+    # Initialize cmd_vel with zero velocity.
+    cmd_vel = zero_twist()
+
+    # Setup for the background event loop thread
+    loop = asyncio.new_event_loop()
+    thread = threading.Thread(target=start_event_loop, args=(loop,), daemon=True)
+    thread.start()
+
+    generated_velocities_future = None
+
+    for event in node:
```

Comment:
@francocipollone meta: it turns out `dora` nodes have support for `async` / `await` patterns, see [this example](https://github.com/dora-rs/dora/blob/66fa54e8a7bca9792f59cce5d53a454f72699c0a/examples/python-async/receive_data.py). It may be cleaner to make it all async rather spinning up a daemon thread like this.